src/runtime/runtime.js

```diff
@@ -518,24 +518,20 @@ export class ConfiguredRuntime {
     /** @private @const {!../components/activities.ActivityPorts} */
     this.activityPorts_ = new ActivityPorts(this.win_);
 
+    //NOTE: 'this' is passed in as a DepsDef.  Do not pass in 'this' before
+    //analytics service and entitlements manager are constructed unless
+    //you are certain they do not rely on them because they are part of that
+    //definition.
+
     /** @private @const {!AnalyticsService} */
     this.analyticsService_ = new AnalyticsService(this);
 
     /** @private @const {!PayClient} */
-    this.payClient_ = new PayClient(
-      this.win_,
-      this.activityPorts_,
-      this.dialogManager_,
-      this.analyticsService_
-    );
+    this.payClient_ = new PayClient(this);
 
     /** @private @const {!Callbacks} */
     this.callbacks_ = new Callbacks();
```

Comment:
Could you move this one so it is just below:

this.activityPorts_ = new ActivityPorts(this.win_);

The reason I prefer it there is that it is a part of DepsDef but doesn't take 'this' as a parameter.  So its slightly cleaner / less risky to declare it earlier.

Reply:
done